css-syntax-3/ident-three-code-points.html

```diff
@@ -0,0 +1,115 @@
+<!doctype html>
+<script src='/resources/testharness.js'></script>
+<script src='/resources/testharnessreport.js'></script>
+<title>Testing valid ident based on first three code points</title>
+<link rel='stylesheet' href='/resources/testharness.css'>
+<link rel='stylesheet' href='../base.css' />
+
+<link rel="author" title="Greg Whitworth" href="gwhit@microsoft.com" />
+<link rel="help" href="https://drafts.csswg.org/css-syntax-3/#would-start-an-identifier" />
+<style>
+    main div {
+      background-color:red;
+    }
+
+    #1 {
+      background-color:green;
+    }
+
+    #-2 {
+      background-color:green;
+    }
+
+    #--3 {
+      background-color:green;
+    }
+
+    #---4 {
+      background-color:green;
+    }
+
+    #a {
+      background-color:green;
+    }
+
+    #-b {
+      background-color:green;
+    }
+
+    #--c {
+      background-color:green;
+    }
+
+    #---d {
+      background-color:green;
+    }
+</style>
+<body>
+    <main>
+        <div id="1" class="item" name="one">test1</div>
+        <div id="-2" class="item" name="two">test2</div>
+        <div id="--3" class="item" name="three">test3</div>
+        <div id="---4" class="item" name="four">test4</div>
+
+        <div id="a" class="item" name="five">test A</div>
+        <div id="-b" class="item" name="six">test B</div>
+        <div id="--c" class="item" name="seven">test C</div>
+        <div id="---d" class="item" name="eight">test D</div>
+    </main>
+
+    <script>
+        var items = document.getElementsByClassName('item');
+
+        generate_tests(assert_equals, [
+            [
+                // One
+                items[0].getAttribute('name') + " should be red",
+                window.getComputedStyle(document.getElementsByClassName('item')[0]).getPropertyValue('background-color'),
+                "rgb(255, 0, 0)"
+            ],
+            [
+                // Two
+                items[1].getAttribute('name') + " should be red",
+                window.getComputedStyle(document.getElementsByClassName('item')[1]).getPropertyValue('background-color'),
+                "rgb(255, 0, 0)"
+            ],
+            [
+                // Three
+                items[2].getAttribute('name') + " should be green",
+                window.getComputedStyle(document.getElementsByClassName('item')[2]).getPropertyValue('background-color'),
+                "rgb(0, 128, 0)"
+            ],
+            [
+                // Four
+                items[3].getAttribute('name') + " should be red",

```

Comment:
> Based on the spec, I believe this is correct but Firefox/Chrome currently allow --- to be a valid ident even though the current spec text only allows two of them (assuming I'm reading it correctly).

That’s not my understanding of the spec.

https://drafts.csswg.org/css-syntax/#check-if-three-code-points-would-start-an-identifier
> Look at the first code point:
>
> U+002D HYPHEN-MINUS
>    If the second code point is a name-start code point or a U+002D HYPHEN-MINUS, or the second and third code points are a valid escape, return true. Otherwise, return false. 

If the first two code points are `--` this returns true regardless of what’s third, including another `-`.